cmd/repo-updater/repos/syncer.go

```diff
@@ -89,69 +138,94 @@ func sleep(ctx context.Context, d time.Duration) {
 	}
 }
 
-// TriggerSync will run Sync now. If a sync is currently running it is
-// cancelled.
-func (s *Syncer) TriggerSync() {
-	s.syncSignal.Trigger()
+// TriggerEnqueueSyncJobs will enqueue any pending sync jobs now.
+func (s *Syncer) TriggerEnqueueSyncJobs() {
+	s.enqueueSignal.Trigger()
 }
 
-// Sync synchronizes the repositories.
-func (s *Syncer) Sync(ctx context.Context) (err error) {
+// SyncExternalService syncs repos using all of the supplied external services
+func (s *Syncer) SyncExternalService(ctx context.Context, store Store, externalServiceID int64, minSyncInterval time.Duration) (err error) {
 	var diff Diff
 
-	ctx, save := s.observe(ctx, "Syncer.Sync", "")
+	log15.Info("Syncing external service", "serviceID", externalServiceID)
```

Comment:
qq: Would it be too verbose with `Info`?
```suggestion
	log15.Debug("Syncing external service", "serviceID", externalServiceID)
```